python/ray/data/tests/test_batcher.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+
+import pyarrow as pa
+
+from ray.data._internal.batcher import ShufflingBatcher
+
+
+def gen_block(num_rows):
+    return pa.table({"foo": [1] * num_rows})
+
```

Comment:
How about adding some parameter validation tests since we have clear contract on the parameters of ShufflingBatcher?

Reply:
Added some simple test cases.